.github/workflows/ci.yml

```diff
@@ -13,27 +26,33 @@ jobs:
       # Install Tools
       - name: Install Tools
         run: |
-          sudo apt-get install wget build-essential python3 verilator libevent-dev libjson-c-dev device-tree-compiler
-          pip3 install setuptools
-          pip3 install requests
-          pip3 install pexpect
+          apt-get -qy update
+          apt-get -qy install wget build-essential python3 python3-pip verilator libevent-dev libjson-c-dev \
+            device-tree-compiler python3-requests python3-pexpect
 
       # Install (n)Migen / LiteX / Cores
       - name: Install LiteX
         run: |
           wget https://raw.githubusercontent.com/enjoy-digital/litex/master/litex_setup.py
-          python3 litex_setup.py init install --user
+          python3 litex_setup.py init install
 
       # Install RISC-V GCC
       - name: Install RISC-V GCC
```

Comment:
This seems to be specific for the testing in this repo. Hence, it might be sensible to have two images. One for Litex, and another one for RISC-V GCC. Ideally, it would not be necessary to have both of them in the same image. See https://hdl.github.io/containers/#_usage.

See also https://github.com/hdl/containers/issues/16. That is, the Litex image might be upstreamed to hdl/containers.